MidasCivil_Adapter/CRUD/Read/Loads/BarDifferentialTemperatureLoad.cs

```diff
@@ -0,0 +1,115 @@
+        /**** Private Methods                           ****/
+        /***************************************************/
+
+        private List<ILoad> ReadBarDifferentialTemperatureLoads(List<string> ids = null)
+        {
+            List<ILoad> bhomBarDifferentialTemperatureLoads = new List<ILoad>();
+
+            List<Loadcase> bhomLoadcases = ReadLoadcases();
+            Dictionary<string, Loadcase> loadcaseDictionary = bhomLoadcases.ToDictionary(
+                        x => x.Name);
+            string[] loadcaseFolders = Directory.GetDirectories(m_directory + "\\TextFiles");
+            int i = 1;
+            foreach (string loadcaseFolder in loadcaseFolders)
+            {
+                string loadcase = Path.GetFileName(loadcaseFolder);
+                List<string> barDifferentialTemperatureLoadText = GetSectionText(loadcase + "\\BSTEMPER");
+                if (barDifferentialTemperatureLoadText.Count != 0)
+                {
+                    List<List<string>> barDifferntialTemperatureSets = new List<List<string>>();
+                    List<string> barID = new List<string>();
+                    List<int> loadIndex = new List<int>();
+                    List<string> delimitted = new List<string>();
+                    for (int j = 0; j < barDifferentialTemperatureLoadText.Count; j++)
+                    {
+                        if (!barDifferentialTemperatureLoadText[j].Contains("ELEMENT"))
+                        {
+                            loadIndex.Add(j);
+                            delimitted = barDifferentialTemperatureLoadText[j].Split(',').ToList();
+                            barID.Add(delimitted[0].Trim());
+                        }
+                    }
+                    loadIndex.Add(barDifferentialTemperatureLoadText.Count);
+                    int index = 0;
+
+                    foreach (int LoadIN in loadIndex)
+                    {
+                        if (LoadIN == loadIndex[0])
+                        {
+                            barDifferntialTemperatureSets.Add(new List<string>());
+                            for (int j = 0; j < loadIndex[1]; j++)
+                            {
+                                barDifferntialTemperatureSets[index].Add(String.Join(",", barDifferentialTemperatureLoadText[j].Split(',').ToList()));
+                            }
+                        }
+                        else if (LoadIN == loadIndex.Last())
+                        {
+                            ;
+                        }
+                        else
+                        {
+                            barDifferntialTemperatureSets.Add(new List<string>());
+                            for (int j = LoadIN; j < loadIndex[1 + index]; j++)
+                            {
+                                barDifferntialTemperatureSets[index].Add(String.Join(",", barDifferentialTemperatureLoadText[j].Split(',').ToList()));
+                            }
```

Comment:
This can be simplified, if you start the `for` loop at the `loadIndex[i]` you can then iterate until `loadIndex[i+1] provided the `foreach` loop above is made in to a `for` loop.

Like this:
`for (int j=loadIndex[i]; j < loadIndex[i+1]; j++)`

This then removes the need for the multiple `if` statements.

Reply:
I have updated it to a `for` loop instead of a `foreach` loop. But I have remained the if check for the final line as the final point of the load index is actually referring to a blank line in the load text file